rllib/agents/trainer.py

```diff
@@ -1607,6 +1608,12 @@ def __getstate__(self) -> dict:
             state["worker"] = self.workers.local_worker().save()
         if hasattr(self, "optimizer") and hasattr(self.optimizer, "save"):
             state["optimizer"] = self.optimizer.save()
+        # TODO: Experimental functionality: Store contents of replay buffer
+        #  to checkpoint, only if user has configured this.
+        if hasattr(self, "_local_replay_buffer") and \
+                self.config.get("store_buffer_in_checkpoints"):
+            state["local_replay_buffer"] = \
```

Comment:
The replay data is already in memory, no? Depending on the buffer's implementation, these could be e.g. lists of SampleBatches. We would just reference the data here from the state-dict.

Reply:
right, right. to clarify a little, there were 2 things I was a bit not sure.
one is when we eventually write this data onto disk, or load it from disk, does our serialization work well for this large chunk of data? we just pickle it right?
two is I was wondering how does this work if we have multiple replay buffer shards? do they have to aggregate all the experiences and write a single file, or each shard would write their own checkpoint files? will there be like a lot of network traffic because of this?

Reply:
Yesm good point about sharded buffers. That's why I left APEX out of this PR for now. We can answer this question in a follow-up PR and cover all replay buffer types then.